### ISSUE TITLE
Hello world fails to build on Windows 10

### ISSUE THREAD

**Issue by mkislik** [awslabs/aws-sam-cli#847]
### Description

Python version of hello-world fails to build on Windows 10

### Steps to reproduce

I can reproduce it in 2 ways.  First way to reproduce:

sam init --runtime=python3.6
cd sam-app
sam build --use-container

Can also be reproduced if I do the following:
Step 0: (if necessary to clean up from above steps, in WSL): cd ..;rm -rf sam-app
sam init --runtime=python3.6
cd sam-app
sam build
sam local start-api --debug

This seems to only happen on my machine, but I don't know how to recover.
I tried to reinstall Python, the AWS CLI, SAM CLI, docker, and I'm having no luck.

### Observed result

PS C:\proj\sls\sam-app> sam local start-api --debug
2018-12-10 12:11:01 Using SAM Template at C:\proj\sls\sam-app\.aws-sam\build\templat
2018-12-10 12:11:01 Changing event name from creating-client-class.iot-data to creat
2018-12-10 12:11:01 Changing event name from before-call.apigateway to before-call.a
2018-12-10 12:11:01 Changing event name from request-created.machinelearning.Predict
.Predict
2018-12-10 12:11:01 Changing event name from before-parameter-build.autoscaling.Crea
meter-build.auto-scaling.CreateLaunchConfiguration
2018-12-10 12:11:01 Changing event name from before-parameter-build.route53 to befor
2018-12-10 12:11:01 Changing event name from request-created.cloudsearchdomain.Searc
ain.Search
2018-12-10 12:11:01 Changing event name from docs.*.autoscaling.CreateLaunchConfigur
o-scaling.CreateLaunchConfiguration.complete-section
2018-12-10 12:11:01 Changing event name from before-parameter-build.logs.CreateExpor
dwatch-logs.CreateExportTask
2018-12-10 12:11:01 Changing event name from docs.*.logs.CreateExportTask.complete-s
ateExportTask.complete-section
2018-12-10 12:11:01 Changing event name from before-parameter-build.cloudsearchdomai
loudsearch-domain.Search
2018-12-10 12:11:01 Changing event name from docs.*.cloudsearchdomain.Search.complet
in.Search.complete-section
2018-12-10 12:11:01 Changing event name from creating-client-class.iot-data to creat
2018-12-10 12:11:01 Changing event name from before-call.apigateway to before-call.a
2018-12-10 12:11:01 Changing event name from request-created.machinelearning.Predict
.Predict
2018-12-10 12:11:01 Changing event name from before-parameter-build.autoscaling.Crea
meter-build.auto-scaling.CreateLaunchConfiguration
2018-12-10 12:11:01 Changing event name from before-parameter-build.route53 to befor
2018-12-10 12:11:01 Changing event name from request-created.cloudsearchdomain.Searc
ain.Search
2018-12-10 12:11:01 Changing event name from docs.*.autoscaling.CreateLaunchConfigur
o-scaling.CreateLaunchConfiguration.complete-section
2018-12-10 12:11:01 Changing event name from before-parameter-build.logs.CreateExpor
dwatch-logs.CreateExportTask
2018-12-10 12:11:01 Changing event name from docs.*.logs.CreateExportTask.complete-s
ateExportTask.complete-section
2018-12-10 12:11:01 Changing event name from before-parameter-build.cloudsearchdomai
loudsearch-domain.Search
2018-12-10 12:11:01 Changing event name from docs.*.cloudsearchdomain.Search.complet
in.Search.complete-section
2018-12-10 12:11:01 local start-api command is called
2018-12-10 12:11:01 Looking for credentials via: env
2018-12-10 12:11:01 Looking for credentials via: assume-role
2018-12-10 12:11:01 Looking for credentials via: shared-credentials-file
2018-12-10 12:11:01 Found credentials in shared credentials file: ~/.aws/credentials
2018-12-10 12:11:01 Loading JSON file: c:\program files\python36\lib\site-packages\b
2018-12-10 12:11:01 Event choose-service-name: calling handler <function handle_serv
>
2018-12-10 12:11:01 Loading JSON file: c:\program files\python36\lib\site-packages\b
08\service-2.json
2018-12-10 12:11:01 Event creating-client-class.serverlessapplicationrepository: cal
presigned_url at 0x0000024E419F7A60>
2018-12-10 12:11:01 The s3 config key is not a dictionary type, ignoring its value o
2018-12-10 12:11:01 Setting serverlessrepo timeout as (60, 60)
2018-12-10 12:11:01 Loading JSON file: c:\program files\python36\lib\site-packages\b
2018-12-10 12:11:01 Registering retry handlers for service: serverlessrepo
2018-12-10 12:11:01 No Parameters detected in the template
2018-12-10 12:11:01 2 resources found in the template
2018-12-10 12:11:01 Found Serverless function with name='HelloWorldFunction' and Cod
2018-12-10 12:11:01 Trying paths: ['C:\\Users\\mkislik\\.docker\\config.json', 'C:\\
2018-12-10 12:11:01 Found file at path: C:\Users\mkislik\.docker\config.json
2018-12-10 12:11:01 Found 'auths' section
2018-12-10 12:11:01 Auth data for https://index.docker.io/v1/ is absent. Client migh
ad.
2018-12-10 12:11:01 Found 'credsStore' section
Traceback (most recent call last):
  File "c:\program files\python36\lib\site-packages\docker\api\client.py", line 140,
    self._custom_adapter = NpipeAdapter(
NameError: name 'NpipeAdapter' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\program files\python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\program files\python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Python36\Scripts\sam.exe\__main__.py", line 9, in <module>
  File "c:\program files\python36\lib\site-packages\click\core.py", line 722, in __c
    return self.main(*args, **kwargs)
  File "c:\program files\python36\lib\site-packages\click\core.py", line 697, in mai
    rv = self.invoke(ctx)
  File "c:\program files\python36\lib\site-packages\click\core.py", line 1066, in in
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\program files\python36\lib\site-packages\click\core.py", line 1066, in in
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\program files\python36\lib\site-packages\click\core.py", line 895, in inv
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\program files\python36\lib\site-packages\click\core.py", line 535, in inv
    return callback(*args, **kwargs)
  File "c:\program files\python36\lib\site-packages\click\decorators.py", line 64, i
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "c:\program files\python36\lib\site-packages\click\core.py", line 535, in inv
    return callback(*args, **kwargs)
  File "c:\program files\python36\lib\site-packages\samcli\commands\local\start_api\
    parameter_overrides)  # pragma: no cover
  File "c:\program files\python36\lib\site-packages\samcli\commands\local\start_api\
    aws_region=ctx.region) as invoke_context:
  File "c:\program files\python36\lib\site-packages\samcli\commands\local\cli_common
enter__
    self._check_docker_connectivity()
  File "c:\program files\python36\lib\site-packages\samcli\commands\local\cli_common
heck_docker_connectivity
    docker_client = docker_client or docker.from_env()
  File "c:\program files\python36\lib\site-packages\docker\client.py", line 85, in f
    timeout=timeout, version=version, **kwargs_from_env(**kwargs)
  File "c:\program files\python36\lib\site-packages\docker\client.py", line 40, in _
    self.api = APIClient(*args, **kwargs)
  File "c:\program files\python36\lib\site-packages\docker\api\client.py", line 145,
    'Install pypiwin32 package to enable npipe:// support'
docker.errors.DockerException: Install pypiwin32 package to enable npipe:// support

### Expected result

Running server I believe on port 3000, returning "hello world".

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS:

Windows 10

2. `sam --version`:

SAM CLI, version 0.8.1

`Add --debug flag to command you are running`

**Comment by mkislik** [comment]
I'm guessing this has something to do with the contents of the Dockerfile you're using when using sam build/local, but I tried to find it in the sam-app folder, but couldn't.

In case it is needed, the output of pip freeze on my PC:

PS C:\proj\sls\sam-app> pip3 freeze
adal==1.2.0
antlr4-python3-runtime==4.7.1
applicationinsights==0.11.7
argcomplete==1.9.4
arrow==0.12.1
asn1crypto==0.24.0
attrs==17.4.0
aws-lambda-builders==0.0.3
aws-sam-cli==0.8.1
aws-sam-translator==1.9.0
awscli==1.16.71
azure-cli==2.0.52
azure-cli-cloud==2.1.0
azure-cli-command-modules-nspkg==2.0.2
azure-cli-configure==2.0.19
azure-cli-core==2.0.52
azure-cli-extension==0.2.3
azure-cli-iot==0.3.4
azure-cli-nspkg==3.0.3
azure-cli-profile==2.1.2
azure-cli-resource==2.1.6
azure-cli-telemetry==1.0.0
azure-common==1.1.16
azure-mgmt-authorization==0.50.0
azure-mgmt-iothub==0.6.0
azure-mgmt-iothubprovisioningservices==0.2.0
azure-mgmt-managementgroups==0.1.0
azure-mgmt-nspkg==3.0.2
azure-mgmt-resource==2.0.0
azure-nspkg==3.0.2
bcrypt==3.1.4
binaryornot==0.4.4
boto3==1.9.57
botocore==1.12.61
bottle==0.12.13
certifi==2018.8.24
cffi==1.11.5
chardet==3.0.4
chevron==0.13.1
click==6.7
colorama==0.3.9
commentjson==0.7.1
cookiecutter==1.6.0
cryptography==2.4.2
dateparser==0.7.0
decorator==4.3.0
docker==3.6.0
docker-pycreds==0.4.0
docutils==0.14
entrypoints==0.2.3
enum-compat==0.0.2
Flask==1.0.2
fstrings==0.1.0
future==0.17.1
humanfriendly==4.17
idna==2.7
iotedgedev==0.81.0
iotedgehubdev==0.6.0
isodate==0.6.0
itsdangerous==0.24
Jinja2==2.10
jinja2-time==0.2.0
jmespath==0.9.3
jsonify==0.5
jsonpath-rw==1.4.0
jsonschema==2.6.0
keyring==17.0.0
knack==0.5.1
MarkupSafe==1.0
msrest==0.6.2
msrestazure==0.4.34
mysql-connector-python==8.0.12
oauthlib==2.1.0
paramiko==2.4.2
ply==3.11
portalocker==1.2.1
poyo==0.4.2
protobuf==3.6.1
pyasn1==0.4.4
pycparser==2.19
Pygments==2.3.0
PyJWT==1.7.1
PyNaCl==1.3.0
pyOpenSSL==18.0.0
pypiwin32==223
pyreadline==2.1
python-dateutil==2.7.5
python-dotenv==0.10.0
pytz==2018.7
pywin32==224
pywin32-ctypes==0.2.0
PyYAML==3.13
regex==2018.11.22
requests==2.20.1
requests-oauthlib==1.0.0
rsa==3.4.2
s3transfer==0.1.13
six==1.11.0
tabulate==0.8.2
typing==3.6.4
tzlocal==1.5.1
urllib3==1.24.1
virtualenv==16.1.0
websocket-client==0.54.0
Werkzeug==0.14.1
whichcraft==0.5.2
PS C:\proj\sls\sam-app>

**Comment by sanathkr** [comment]
This is something to do with Docker on Windows. The real error is:

`docker.errors.DockerException: Install pypiwin32 package to enable npipe:// support`

I have observed this before, but don't quite remember what the solution was. It was either to disconnect from VPN, or restart Docker or re-install Docker.

**Comment by mkislik** [comment]
And, as previously noted, pip freeze shows:
...
pypiwin32==223

So the problem seems to be either my global docker image not containing pypiwin32, or SAM's docker image not containing pypiwin32, and I'm not sure how to go about modifying either.

**Comment by mkislik** [comment]
Also the runtime I choose for sam-init doesn't matter.  I still get the same error when I do:
sam init --runtime=nodejs8.10
sam build --use-container

I still get:
  File "c:\program files\python36\lib\site-packages\docker\api\client.py", line 140, in __init__
    self._custom_adapter = NpipeAdapter(
NameError: name 'NpipeAdapter' is not defined
...
'Install pypiwin32 package to enable npipe:// support'

**Comment by TheSriram** [comment]
How have you installed sam cli? can you try installing with the MSI and giving it a shot, and removing the pip installed aws-sam-cli.

I have seen this error show up, but I remember that it was docker being wonky.

**Comment by mkislik** [comment]
Yes, I have installed aws-sam-cli

**Comment by TheSriram** [comment]
Yes node is not yet supported for build, but is in the pipeline: https://github.com/awslabs/aws-lambda-builders/commits/develop

**Comment by TheSriram** [comment]
Try installing aws-sam-cli through our dedicated installer : [64 bit MSI](https://github.com/awslabs/aws-sam-cli/releases/download/v0.8.1/AWS_SAM_CLI_64_PY3.msi). It looks like there maybe multiple python/package versions at play, not allowing for isolation in your environment. also when I checked the dist version in the installed artifacts pypiwin32, it was at 224.

**Comment by mkislik** [comment]
I installed the 64 bit MSI, but I'm not changing pypiwin32==220
The reason is that
1. I don't see a version 224:
https://pypi.org/project/pypiwin32/#history
2. I fail to install it, and I see the following message:
PS C:\proj\sls\sam-app> pip3 install pypiwin32==224
Collecting pypiwin32==224
  Could not find a version that satisfies the requirement pypiwin32==224 (from versions: 219, 220, 223)
No matching distribution found for pypiwin32==224

**Comment by TheSriram** [comment]
Apologies, it was pywin32 that was at 224. pypiwin32 is at 223.

**Comment by mkislik** [comment]
pypiwin32 version 223 was the version causing me problems.
For the purpose of running "sam local" and "sam build -u", and on Windows, you need version 220.

**Comment by mpiroc** [comment]
I see this issue on Windows, and the workaround (installing `pypiwin32==220`) does not make a difference for me.

* Affects at least python and nodejs sam apps
* Affects at least SAM CLI 0.9 and 0.10
* Affects at least python 3.6.5 and 3.6.8

Output of `python -m pip freeze`:

```
PS C:\Users\pirocchi> python --version
Python 3.6.8
PS C:\Users\pirocchi> python -m pip freeze
arrow==0.12.1
astroid==2.0.4
aws-sam-translator==1.6.0
awscli==1.15.26
binaryornot==0.4.4
boto3==1.7.40
botocore==1.10.40
certifi==2018.4.16
chardet==3.0.4
click==6.7
colorama==0.3.9
cookiecutter==1.6.0
dateparser==0.7.0
docker==3.4.0
docker-pycreds==0.3.0
docutils==0.14
enum34==1.1.6
Flask==1.0.2
future==0.16.0
idna==2.7
isort==4.3.4
itsdangerous==0.24
Jinja2==2.10
jinja2-time==0.2.0
jmespath==0.9.3
jsonschema==2.6.0
lazy-object-proxy==1.3.1
MarkupSafe==1.0
mccabe==0.6.1
poyo==0.4.1
pyasn1==0.4.3
pylint==2.1.1
pypiwin32==220
pystache==0.5.4
python-dateutil==2.7.3
pytz==2018.5
PyYAML==3.12
regex==2018.8.29
requests==2.19.1
rsa==3.4.2
s3transfer==0.1.13
six==1.11.0
typed-ast==1.1.0
tzlocal==1.5.1
urllib3==1.23
websocket-client==0.48.0
Werkzeug==0.14.1
whichcraft==0.4.1
wrapt==1.10.11
```

**Comment by mkislik** [comment]
I'm using Python 3.6.7, and it works for me and another colleague.  I know the fix of using pypiwin32=220 does NOT work with Python3.7

Here's my Python setting:
C:\WINDOWS\system32>python --version
Python 3.6.7

C:\WINDOWS\system32>pip3 freeze
adal==1.2.0
antlr4-python3-runtime==4.7.1
applicationinsights==0.11.7
argcomplete==1.9.4
arrow==0.12.1
asn1crypto==0.24.0
attrs==17.4.0
aws-lambda-builders==0.0.3
aws-sam-cli==0.8.1
aws-sam-translator==1.9.0
awscli==1.16.71
azure-cli==2.0.52
azure-cli-cloud==2.1.0
azure-cli-command-modules-nspkg==2.0.2
azure-cli-configure==2.0.19
azure-cli-core==2.0.52
azure-cli-extension==0.2.3
azure-cli-iot==0.3.4
azure-cli-nspkg==3.0.3
azure-cli-profile==2.1.2
azure-cli-resource==2.1.6
azure-cli-telemetry==1.0.0
azure-common==1.1.16
azure-mgmt-authorization==0.50.0
azure-mgmt-iothub==0.6.0
azure-mgmt-iothubprovisioningservices==0.2.0
azure-mgmt-managementgroups==0.1.0
azure-mgmt-nspkg==3.0.2
azure-mgmt-resource==2.0.0
azure-nspkg==3.0.2
bcrypt==3.1.4
beautifulsoup4==4.7.1
binaryornot==0.4.4
boto3==1.9.57
botocore==1.12.61
bottle==0.12.13
bs4==0.0.1
certifi==2018.8.24
cffi==1.11.5
chardet==3.0.4
chevron==0.13.1
click==6.7
colorama==0.3.9
commentjson==0.7.1
cookiecutter==1.6.0
cryptography==2.4.2
dateparser==0.7.0
decorator==4.3.0
docker==3.6.0
docker-pycreds==0.4.0
docutils==0.14
dpkt==1.9.2
entrypoints==0.2.3
enum-compat==0.0.2
Flask==1.0.2
fstrings==0.1.0
future==0.17.1
humanfriendly==4.17
idna==2.7
iotedgedev==0.81.0
iotedgehubdev==0.6.0
isodate==0.6.0
itsdangerous==0.24
Jinja2==2.10
jinja2-time==0.2.0
jmespath==0.9.3
jsonify==0.5
jsonpath-rw==1.4.0
jsonschema==2.6.0
keyring==17.0.0
knack==0.5.1
MarkupSafe==1.0
msrest==0.6.2
msrestazure==0.4.34
mysql-connector-python==8.0.12
oauthlib==2.1.0
paramiko==2.4.2
ply==3.11
portalocker==1.2.1
poyo==0.4.2
prettytable==0.7.2
protobuf==3.6.1
pyasn1==0.4.4
pychrome==0.2.2
pycparser==2.19
Pygments==2.3.0
PyJWT==1.7.1
PyNaCl==1.3.0
pyOpenSSL==18.0.0
pyping==0.0.5
pypiwin32==220
pyreadline==2.1
python-dateutil==2.7.5
python-dotenv==0.10.0
pytz==2018.7
pywin32==224
pywin32-ctypes==0.2.0
PyYAML==3.13
regex==2018.11.22
requests==2.20.0
requests-oauthlib==1.0.0
requests-toolbelt==0.8.0
rsa==3.4.2
s3transfer==0.1.13
selenium==3.141.0
six==1.11.0
soupsieve==1.7.2
tabulate==0.8.2
tcping==0.1.1rc1
typing==3.6.4
tzlocal==1.5.1
urllib3==1.24.1
virtualenv==16.1.0
websocket-client==0.54.0
Werkzeug==0.14.1
whichcraft==0.5.2

**Comment by mpiroc** [comment]
@mkislik Thanks! It looks like this issue was caused by an out-of-date version of the 'docker' pip package.

None of the following had any effect:

1. Remove python 3.6.8 and install python 3.6.7.
2. Update the `awscli` package: `python -m pip install --upgrade awscli`
3. Install the `aws-sam-cli` package: `python -m pip install aws-sam-cli`
4. Restart my machine.

But this fixed the issue:

1. Update the `docker` package from 3.4.0 to 3.7.0: `python -m pip install --upgrade docker`


```
PS C:\repos\t1> python --version
Python 3.6.7
PS C:\repos\t1> python -m pip freeze
arrow==0.12.1
astroid==2.0.4
aws-lambda-builders==0.0.5
aws-sam-cli==0.10.0
aws-sam-translator==1.9.0
awscli==1.16.93
binaryornot==0.4.4
boto3==1.9.83
botocore==1.12.83
certifi==2018.4.16
chardet==3.0.4
chevron==0.13.1
click==6.7
colorama==0.3.9
cookiecutter==1.6.0
dateparser==0.7.0
docker==3.7.0
docker-pycreds==0.4.0
docutils==0.14
enum34==1.1.6
Flask==1.0.2
future==0.16.0
idna==2.7
isort==4.3.4
itsdangerous==0.24
Jinja2==2.10
jinja2-time==0.2.0
jmespath==0.9.3
jsonschema==2.6.0
lazy-object-proxy==1.3.1
MarkupSafe==1.0
mccabe==0.6.1
poyo==0.4.1
pyasn1==0.4.3
pylint==2.1.1
pypiwin32==223
pystache==0.5.4
python-dateutil==2.7.3
pytz==2018.5
pywin32==224
PyYAML==3.12
regex==2018.8.29
requests==2.20.1
rsa==3.4.2
s3transfer==0.1.13
serverlessrepo==0.1.5
six==1.11.0
typed-ast==1.1.0
tzlocal==1.5.1
urllib3==1.23
websocket-client==0.48.0
Werkzeug==0.14.1
whichcraft==0.4.1
wrapt==1.10.11
PS C:\repos\t1> sam build
2019-01-23 15:52:39 Found credentials in shared credentials file: ~/.aws/credentials
2019-01-23 15:52:39 Building resource 'HelloWorldFunction'
2019-01-23 15:52:39 'nodejs' runtime has not been validated!
2019-01-23 15:52:39 Running NodejsNpmBuilder:NpmPack
2019-01-23 15:52:40 Running NodejsNpmBuilder:CopySource
2019-01-23 15:52:40 Running NodejsNpmBuilder:NpmInstall

Build Succeeded

Built Artifacts  : .aws-sam\build
Built Template   : .aws-sam\build\template.yaml

Commands you can use next
=========================
[*] Invoke Function: sam local invoke
[*] Package: sam package --s3-bucket <yourbucket>

PS C:\repos\t1> echo '{}' | sam local invoke HelloWorldFunction
2019-01-23 15:52:48 Reading invoke payload from stdin (you can also pass it from file with --event)
2019-01-23 15:52:48 Found credentials in shared credentials file: ~/.aws/credentials
2019-01-23 15:52:48 Invoking app.lambdaHandler (nodejs8.10)

Fetching lambci/lambda:nodejs8.10 Docker container image......
2019-01-23 15:52:49 Mounting C:\repos\t1\.aws-sam\build\HelloWorldFunction as /var/task:ro inside runtime container
START RequestId: f80782ca-3f07-163f-9e20-efc980427f3c Version: $LATEST
Unable to import module 'app': Error
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
END RequestId: f80782ca-3f07-163f-9e20-efc980427f3c
REPORT RequestId: f80782ca-3f07-163f-9e20-efc980427f3c  Duration: 2.35 ms       Billed Duration: 100 ms Memory Size: 128 MB     Max Memory Used: 31 MB

{"errorMessage":"Cannot find module '/var/task/app'","errorType":"Error","stackTrace":["Function.Module._load (module.js:474:25)","Module.require (module.js:596:17)","require (internal/module.js:11:18)"]}
```

I'm investigating the new error now, but it looks like an unrelated issue.

So the question is: Why was my docker dependency out of date despite just installing SAM CLI 0.10 yesterday? Should the minimum version be updated in SAM CLI's dependencies?

**Comment by mpiroc** [comment]
I think you just need to update the docker dependency [here](https://github.com/awslabs/aws-sam-cli/blob/develop/requirements/base.txt) to at least 3.6.0 (3.5.0 might work as well. 3.4.0 is too early).

**Comment by mkislik** [comment]
Don't know about your error, but doing:
python -m pip install --upgrade docker
as you've suggested installs pypiwin32==224, which causes the NPIPE issue I reported in this bug. 
And:
python -m pip install pypiwin32==220 fixes the issue.

**Comment by awood45** [comment]
This has been inactive for a long time and is believed to be solved at this point.

If this issue is recurring or if there are more details, feel free to reopen with the new issues/details.

**Comment by carbonox-infernox** [comment]
@awood45 Would you mind reopening this for me? I don't seem to have the option.

**Comment by maxkislik** [comment]
I don't see the option to reopen this issue.

I remember working around the issue by: downgrading pypiwin32=220, but that was around 10 months ago.  Others are saying downgrading pywin32 from version 225 to 224 worked for them.

I also asked AWS to document this problem and the solution, but that did not happen, either.

**Comment by awood45** [comment]
I can reopen, but how are y'all installing the AWS SAM CLI? Are you using the MSI, or something else?

**Comment by maxkislik** [comment]
Hi Alex,



It has been a while, but I believe I used pip to install aws-cli



Max



*From:* Alex Wood <notifications@github.com>
*Sent:* Thursday, October 31, 2019 10:44 AM
*To:* awslabs/aws-sam-cli <aws-sam-cli@noreply.github.com>
*Cc:* maxkislik <max.kislik@nomadix.com>; Comment <
comment@noreply.github.com>
*Subject:* Re: [awslabs/aws-sam-cli] Hello world fails to build on Windows
10 (#847)



I can reopen, but how are y'all installing the AWS SAM CLI? Are you using
the MSI, or something else?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/awslabs/aws-sam-cli/issues/847?email_source=notifications&email_token=ANMQUXV5OOAK6RWUR2Y5HPDQRMKN3A5CNFSM4GJQMRNKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECYU4RA#issuecomment-548490820>,
or unsubscribe

**Comment by carbonox-infernox** [comment]
I also used pip to install aws-sam-cli

**Comment by BhayanakMoth** [comment]
So I installed sam-cli using msi and I was getting this issue.
I tried everything in this thread and reinstalled everything, but the issue persisted nevertheless.   

It turns out sam-cli is using a different python installation at runtime, and for me, it is located in   
```C:\Program Files\Amazon\AWSSAMCLI\runtime```   
This thing uses docker-4.0.4 which is outdated so I ran ```./python -m pip install --upgrade docker``` in the runtime folder, and after upgrading to docker-5.0.3 I am longer facing the issue.
